gui/kitchen-info.lua

```diff
@@ -0,0 +1,414 @@
+                    local bag_mat_type  = reaction_product.material.mat_type[i]
+                    local bag_mat_index = reaction_product.material.mat_index[i]
+                    local bag_matinfo = dfhack.matinfo.decode(bag_mat_type, bag_mat_index)
+                    local bag_mat_plant_raw = bag_matinfo.plant
+                    local bag_mat_plant_growth = bag_mat_plant_raw.growths[bag_item_subtype]
+
+                    if (bag_mat_plant_growth.name_plural:sub(-6) == "leaves") then
+                        return "Leaves"
+                    end
+                end
+
+                return "Other"
+            end
+        end
+    end
+    return false
+end
+--]]
+
+function COLUMNS_BY_LABEL.Thread.getColumnValue(item_type, material, plant_raw)
+    if (item_type == df.item_type.PLANT and plant_raw.flags.THREAD) then
+        return "Thread"
+    end
+    return false
+end
+
+function COLUMNS_BY_LABEL.Mill.getColumnValue(item_type, material, plant_raw)
+    if (item_type == df.item_type.PLANT and plant_raw.flags.MILL) then
+        local mill_mat_type = plant_raw.material_defs.type.mill
+        local mill_mat_index = plant_raw.material_defs.idx.mill
+        local mill_matinfo = dfhack.matinfo.decode(mill_mat_type, mill_mat_index)
+        local mill_material = mill_matinfo.material
+
+        if (mill_material.flags.IS_DYE) then
+            return "Dye"
+        elseif (mill_material.state_name.Powder:sub(-5) == "flour") then
+            return "Flour"
+        else
+            return "Other"
+        end
+    elseif (item_type == df.item_type.SEEDS) then
+        for _, reaction_product_id in ipairs(material.reaction_product.id) do
+            if (reaction_product_id.value == "PRESS_LIQUID_MAT") then
+                return "Paste"
+            end
+        end
+    end
+    return false
+end
+
+--[[
+-- TODO?
+function COLUMNS_BY_LABEL.Press.getColumnValue(item_type, material, plant_raw)
+    return false
+end
+--]]
+
+function COLUMNS_BY_LABEL.EatRaw.getColumnValue(item_type, material, plant_raw)
+    if (material.flags.EDIBLE_RAW) then
+        return "Eat"
+    end
+    return false
+end
+
+local function findStringInLine(x1, x2, y, str)
+    local length = str:len()
+    local firstByte = str:byte(1)
+    local xEnd = x2 - (length - 1)
+    for x = x1, xEnd do
+        if (dfhack.screen.readTile(x, y).ch == firstByte) then
+            local matches = true
+            for i = 2, length do
+                if (dfhack.screen.readTile(x + i - 1, y).ch ~= str:byte(i)) then
+                    matches = false
+                    break
+                end
+            end
+            if (matches) then
+                return x, x + length - 1
+            end
+        end
+    end
+    return -1, -1
+end
+
+local function drawNumber(x, kitchen, firstVisibleIndex, numDisplayedItems)
+    dfhack.screen.paintString({ fg = COLOR_WHITE }, x, 4, "Number")
```

Comment:
"Number" is a bit ambiguous. This is the number of items in stock, right? How about calling it "Stock" or "Count"?

Reply:
But the actual column in DF is labeled "Number".
I will call it "drawNumberColumn".